async-opcua-codegen/src/input/nodeset.rs

```diff
@@ -0,0 +1,171 @@
+        if let Some(attr) = obj.attributes.get("xmlns") {
+            map.insert(attr.clone());
+        }
+        if let Some(attr) = obj.attributes.get("xmlns:uax") {
+            map.insert(attr.clone());
+        }
+        for child in obj.children.values() {
+            for child in child {
+                Self::find_referenced_xsd_schemas_rec(child, map);
+            }
+        }
+    }
+
+    fn find_referenced_xsd_schemas_variant(variant: &Variant, map: &mut HashSet<String>) {
+        match variant {
+            opcua_xml::schema::opc_ua_types::Variant::ExtensionObject(obj) => {
+                if let Some(body) = obj.body.as_ref().and_then(|b| b.data.as_ref()) {
+                    Self::find_referenced_xsd_schemas_rec(body, map);
+                }
+            }
+            opcua_xml::schema::opc_ua_types::Variant::ListOfExtensionObject(objs) => {
+                for obj in objs {
+                    if let Some(body) = obj.body.as_ref().and_then(|b| b.data.as_ref()) {
+                        Self::find_referenced_xsd_schemas_rec(body, map);
+                    }
+                }
+            }
+            opcua_xml::schema::opc_ua_types::Variant::Variant(variant) => {
+                Self::find_referenced_xsd_schemas_variant(variant, map);
+            }
+            _ => (),
+        }
+    }
+
+    fn find_referenced_xsd_schemas(node_set: &UANodeSet) -> HashSet<String> {
+        // Recursively look through all values to find which XSD schemas are referenced,
+        // since this isn't reported anywhere centrally.
+        let mut res = HashSet::new();
+        for node in &node_set.nodes {
+            let value = match node {
+                opcua_xml::schema::ua_node_set::UANode::Variable(v) => v.value.as_ref(),
+                opcua_xml::schema::ua_node_set::UANode::VariableType(v) => v.value.as_ref(),
+                _ => continue,
+            };
+            let Some(value) = value else {
+                continue;
+            };
+            Self::find_referenced_xsd_schemas_variant(&value.0, &mut res);
+        }
+        res
+    }
+
+    pub fn parse(data: &str, path: &str, docs: Option<&str>) -> Result<Self, CodeGenError> {
+        let nodeset = load_nodeset2_file(data)?;
+
+        let Some(nodeset) = nodeset.node_set else {
+            return Err(CodeGenError::missing_required_value("NodeSet"));
+        };
+        let aliases = nodeset.aliases.as_ref().map(|a| {
+            a.aliases
+                .iter()
+                .map(|a| (a.alias.clone(), a.id.0.clone()))
+                .collect::<HashMap<_, _>>()
+        });
+        let Some(models) = nodeset.models.as_ref() else {
+            return Err(CodeGenError::missing_required_value("Models"));
+        };
+
+        if models.models.len() > 1 {
+            println!("Warning, multiple models found in nodeset file, this is not supported, and only the first will be used.");
```

Comment:
Do you intended these to be `println`, not `log::debug` (or maybe `log::info`)? Noticed this was in the code already, but it just caught my eye...